esmvaltool/diag_scripts/land_carbon_cycle/diag_global_turnover.py

```diff
@@ -0,0 +1,822 @@
+    colors1g = np.vstack((colors1, colorsgr))
+    colors = np.vstack((colors1g, colors2))
+    cb_info_ratio['colMap'] = mpl.colors.LinearSegmentedColormap.from_list(
+        'my_colormap', colors)
+    cb_info_ratio['ticksLoc'] = [0.2, 0.25, 0.33, 0.5, 0.9, 1.1, 2, 3, 4, 5]
+    cb_info_ratio['ticksLab'] = [
+        '$\\dfrac{1}{5}$', '$\\dfrac{1}{4}$', '$\\dfrac{1}{3}$',
+        '$\\dfrac{1}{2}$', '$\\dfrac{1}{1.1}$', '$1.1$', '$2$', '$3$', '$4$',
+        '$5$'
+    ]
+    return cb_info_ratio
+
+
+# data and calculation functions
+
+
+def _apply_common_mask(dat_1, dat_2):
+    '''
+    apply a common mask of valid grid cells across two input arrays
+    '''
+    dat_1_mask = np.ma.getmask(np.ma.masked_invalid(dat_1))
+    dat_2_mask = np.ma.getmask(np.ma.masked_invalid(dat_2))
+    _val_mask_a = 1 - (1 - dat_1_mask) * (1 - dat_2_mask)
+    _val_mask = np.ma.nonzero(_val_mask_a)
+    dat_1[_val_mask] = np.nan
+    dat_2[_val_mask] = np.nan
+    dat_1 = np.ma.masked_invalid(dat_1)
+    dat_2 = np.ma.masked_invalid(dat_2)
+    return dat_1, dat_2
+
+
+def _get_agreement_mask(mmdat, dat_5, dat_95, fig_config, nmodel_reject=2):
+    '''
+    get the mask of regiosn where fewer than one quarter of the model
+    simulations are outside the range of observational uncertainty
+    '''
+    _maskf = np.zeros_like(mmdat)
+    _maskf[(mmdat < dat_95) & (mmdat > dat_5)] = 1
+    num_count = _maskf.sum(0)
+    agreement_mask = np.zeros_like(num_count)
+    agreement_mask[num_count < nmodel_reject] = 1
+    wnan = np.ma.masked_invalid(dat_5).mask
+    agreement_mask[wnan] = 0.
+    return agreement_mask
+
+
+def _get_hex_data(dat_1, dat_2, fig_config):
+    '''
+    get the data to be plotted as density plots, which requires that both the
+    arrays have the same mask with regards to valid data points
+    '''
+    dat_1[(dat_1 < fig_config['valrange_sc'][0] * 0.5)] = np.nan
+    dat_1[(dat_1 > fig_config['valrange_sc'][1] * 1.5)] = np.nan
+    dat_2[(dat_2 < fig_config['valrange_sc'][0] * 0.5)] = np.nan
+    dat_2[(dat_2 > fig_config['valrange_sc'][1] * 1.5)] = np.nan
+    dat_1, dat_2 = _apply_common_mask(dat_1, dat_2)
+    dat_1mc = np.ma.masked_equal(dat_1, np.nan).compressed()
+    dat_2mc = np.ma.masked_equal(dat_2, np.nan).compressed()
+    return dat_1mc, dat_2mc
+
+
+def _get_obs_data(diag_config):
+    '''
+    Get and handle the observations of turnover time from Carvalhais 2014.
+
+    Arguments:
+        diag_config - nested dictionary of metadata
+
+    Returns:
+        dictionary with observation data with different variables as keys
+    '''
+    if not diag_config.get('obs_variable'):
+        raise ValueError('The observation variable needs to be specified in '
+                         'the recipe (see recipe description for details)')
+    else:
+        obs_dir = os.path.join(diag_config['auxiliary_data_dir'],
+                               diag_config['obs_info']['obs_data_subdir'])
+
```

Comment:
This has been discussed with @zklaus. Unfortunately, the people at MPI BGC want to keep it in house so that updates can reach every user. The data is easily available in a few minutes, and prepared in obs4mip standard.

Reply:
still, this bit here can be ported to a cmorization script that an ESMValTool user can run to cmorize their freshly downloaded data. Don't worry about it now, but I think it might be useful for the future